ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -76,35 +89,30 @@ Control {
     QtObject {
         id: d
 
-        function setSelectedHoldingId(holdingId, holdingType) {
-            let holding = SQUtils.ModelUtils.getByKey(root.processedAssetsModel, "symbol", holdingId)
-            d.selectedHoldingId = holdingId
-            d.setSelectedHolding(holding, holdingType)
-        }
-
-        function setSelectedHolding(holding, holdingType) {
-            d.selectedHoldingType = holdingType
-            d.selectedHolding = holding
-            holdingSelector.setSelectedItem(holding, holdingType)
-        }
+        property var selectedHolding: SQUtils.ModelUtils.getByKey(holdingSelector.model, "tokensKey", holdingSelector.currentTokensKey)
 
-        property var selectedHolding: null
-        property var selectedHoldingType: Constants.TokenType.Unknown
-        property string selectedHoldingId
-
-        readonly property bool isSelectedHoldingValidAsset: !!selectedHolding && selectedHoldingType === Constants.TokenType.ERC20
-        readonly property double maxFiatBalance: isSelectedHoldingValidAsset ? selectedHolding.currentCurrencyBalance : 0
+        readonly property bool isSelectedHoldingValidAsset: !!selectedHolding
+        readonly property double maxFiatBalance: isSelectedHoldingValidAsset ? selectedHolding.currencyBalance : 0
         readonly property double maxCryptoBalance: isSelectedHoldingValidAsset ? selectedHolding.currentBalance : 0
         readonly property double maxInputBalance: amountToSendInput.inputIsFiat ? maxFiatBalance : maxCryptoBalance
```

Comment:
I was wondering if we can proactively remove all doubles from here already? and under the https://github.com/status-im/status-desktop/issues/15207 the base component will get fixed? also involves MaxButton which uses doubles?

Reply:
Besides, it has really nothing to do with the topic of this PR/issue (TokenSelector); this is about the input component/handling